graylog2-server/src/main/java/org/graylog2/rest/resources/system/indexer/IndexSetDefaultsResource.java

```diff
@@ -0,0 +1,85 @@
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * Server Side Public License for more details.
+ *
+ * You should have received a copy of the Server Side Public License
+ * along with this program. If not, see
+ * <http://www.mongodb.com/licensing/server-side-public-license>.
+ */
+package org.graylog2.rest.resources.system.indexer;
+
+import com.codahale.metrics.annotation.Timed;
+import io.swagger.annotations.Api;
+import io.swagger.annotations.ApiOperation;
+import io.swagger.annotations.ApiParam;
+import org.apache.shiro.authz.annotation.RequiresAuthentication;
+import org.apache.shiro.authz.annotation.RequiresPermissions;
+import org.graylog2.audit.AuditEventTypes;
+import org.graylog2.audit.jersey.AuditEvent;
+import org.graylog2.configuration.IndexSetsDefaultConfiguration;
+import org.graylog2.indexer.IndexSetValidator;
+import org.graylog2.plugin.cluster.ClusterConfigService;
+import org.graylog2.shared.rest.resources.RestResource;
+import org.graylog2.shared.security.RestPermissions;
+import org.joda.time.Duration;
+
+import javax.inject.Inject;
+import javax.validation.constraints.NotNull;
+import javax.ws.rs.BadRequestException;
+import javax.ws.rs.Consumes;
+import javax.ws.rs.PUT;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import java.io.IOException;
+
+@RequiresAuthentication
+@Api(value = "System/IndexSetDefaults", description = "Index set defaults")
+@Path("/system/indices/index_set_defaults")
+@Produces(MediaType.APPLICATION_JSON)
+public class IndexSetDefaultsResource extends RestResource {
+    private final IndexSetValidator indexSetValidator;
+    private final ClusterConfigService clusterConfigService;
+
+    @Inject
+    public IndexSetDefaultsResource(IndexSetValidator indexSetValidator, ClusterConfigService clusterConfigService) {
+        this.indexSetValidator = indexSetValidator;
+        this.clusterConfigService = clusterConfigService;
+    }
+
+    /**
+     * Save new {@link IndexSetsDefaultConfiguration} cluster configuration object. This method exists to allow additional validation
+     * before saving with the {@link ClusterConfigService}.
+     */
+    @PUT
+    @Timed
+    @Consumes(MediaType.APPLICATION_JSON)
+    @ApiOperation(value = "Update index set defaults configuration")
+    @RequiresPermissions({RestPermissions.CLUSTER_CONFIG_ENTRY_CREATE, RestPermissions.CLUSTER_CONFIG_ENTRY_EDIT})
+    @AuditEvent(type = AuditEventTypes.CLUSTER_CONFIGURATION_UPDATE)
+    public Response update(@ApiParam(name = "body", value = "The payload of the index set defaults configuration", required = true)
+                           @NotNull IndexSetsDefaultConfiguration config) throws IOException {
+        IndexSetValidator.Violation violation = indexSetValidator.validateRefreshInterval(
```

Comment:
Playing around with this I think it may be a good idea to beef up the default index config field validation. For instance it's possible to save negative values for many of the numerical fields as well as invalid analyzer types.

![image](https://user-images.githubusercontent.com/103449971/200899404-c91d77c8-3875-43f7-a122-7d4854036592.png)

Some of the bad numerical values are disallowed on the actual Create Index page (by a combo of front end and back end validation) stopping actual index creation there, but the Illuminate installer will still create indices with the bad config values.

![image](https://user-images.githubusercontent.com/103449971/200899195-7e73a180-339e-4f8c-a465-ba7620eeae95.png)

Reply:
Good call! I will get this validation in place tomorrow morning on both frontend and backend.

Reply:
These updates look solid. I think you're right that it makes sense to address the leftover validation deficiencies together in another PR as this is already pretty expansive.  

The list you have covers the other validation I'm tracking. If users can create their own analyzers then then definitely complicates validation of that field, and further supports investigating later in another PR.